packages/database/src/migrations/20210125042225-SamoaCovidSurveyHierarchyUpdate-modifies-data.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+import { updateValues } from '../utilities/migration';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const addressIdQuestionCode = 'QMIA029';
+const addressTextQuestionCode = 'QMIA010';
+const caseQuestionCode = 'QMIA025';
+const samoaEntityId = '5df1b88c61f76a485cd1ca09';
+const surveyResponseId = 'survey_response_id';
+
+async function getAnswersByQuestionCode(db, code) {
+  return db.runSql(`
+        select a.*
+        from answer a
+        inner join survey_response sr
+        on a.survey_response_id = sr.id
+        inner join question q
+        on a.question_id = q.id
+        where q.code in ('${code}');    
+  `);
+}
+
+async function findEntityIdByName(db, name) {
+  const cleanName = name.replace("'", '').trim();
+  if (!name.includes('/'))
+    return db.runSql(`
+  select * from entity
+  where name = '${cleanName}'
+`);
+  return null;
+}
+
+async function changeEntityCaseParentToVillage(db, caseParentMapping) {
+  Promise.all(
+    caseParentMapping.map(async cas => {
+      const [caseCode, parentId] = Object.entries(cas)[0];
+      if (parentId) await updateValues(db, 'entity', { parent_id: parentId }, { code: caseCode });
+    }),
+  );
+}
+
+exports.up = async function (db) {
+  const cases = (await getAnswersByQuestionCode(db, caseQuestionCode)).rows;
+  const addressesIds = (await getAnswersByQuestionCode(db, addressIdQuestionCode)).rows;
+  const addressesTexts = (await getAnswersByQuestionCode(db, addressTextQuestionCode)).rows;
+
+  const caseParentMapping = await Promise.all(
+    cases.map(async cas => {
```

Comment:
One extra letter is worth the readability:

```suggestion
    cases.map(async case => {
```

Reply:
It will have conflict with `case` from `switch`. Perhaps `individual`?

Reply:
Oh it's a reserved word!! Haha interesting. I'd suggest something like `currentCase`